tasks/eva_2038/storeSSIDsInVariantWarehouse.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import logging
+import os
+import random
+import shutil
+import string
+import subprocess
+
+import yaml
+from ebi_eva_common_pyutils import command_utils
+from ebi_eva_common_pyutils.logger import logging_config as log_cfg
+from ebi_eva_internal_pyutils.metadata_utils import get_metadata_connection_handle, resolve_variant_warehouse_db_name
+from ebi_eva_internal_pyutils.pg_utils import get_all_results_for_query
+
+logger = log_cfg.get_logger(__name__)
+log_cfg.add_stdout_handler()
+log_cfg.set_log_level(logging.INFO)
+
+
+def run_nextflow(working_dir, params, project_id, accession_report_file, db_name):
+    random_string = ''.join(random.choice(string.ascii_letters) for i in range(6))
+    workflow_name = f"import_accession_{project_id}_{random_string}"
+
+    # create nextflow work dir - remove if already exists
+    nextflow_work_dir = os.path.join(working_dir, project_id, workflow_name)
+    if os.path.exists(nextflow_work_dir):
+        shutil.rmtree(nextflow_work_dir)
+    os.makedirs(nextflow_work_dir)
+
+    try:
+        command_utils.run_command_with_output(
+            f'Nextflow {workflow_name} process',
+            ' '.join((
+                'export NXF_OPTS="-Xms1g -Xmx8g"; ',
+                params['nextflow_path'], params['nextflow_script'],
+                '-work-dir', nextflow_work_dir,
+                '--java_app', params['java_app'],
+                '--acc_import_job_props', params['acc_import_job_props'],
+                '--accession_report', accession_report_file,
+                '--db_name', db_name,
+                '--logs_dir', os.path.join(working_dir, project_id)
+            ))
+        )
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Nextflow {workflow_name} pipeline failed: results might not be complete.')
+        raise e
+
+
+def get_db_name(params, project_id, accession_report_file):
+    with get_metadata_connection_handle("production_processing", params['private_settings_xml_file']) as pg_conn:
+        query = f"""
+            select distinct asm.assembly_accession, t.taxonomy_id
+            from project_taxonomy pt 
+            join taxonomy t on t.taxonomy_id = pt.taxonomy_id 
+            join project_analysis pa on pa.project_accession = pt.project_accession 
+            join analysis a on pa.analysis_accession = a.analysis_accession 
+            join assembly asm on asm.assembly_accession = a.vcf_reference_accession 
+            join analysis_file af on af.analysis_accession = pa.analysis_accession 
+            join file f on f.file_id = af.file_id 
+            where pt.project_accession = '{project_id}' 
+            and f.filename = '{os.path.basename(accession_report_file).replace(".accessioned.vcf", ".vcf")}'
+            and f.file_type IN ('vcf', 'VCF', 'vcf_aggregate', 'VCF_AGGREGATE')
+            """
```

Comment:
This query is probably fine, but I think you can save some joins by using the `assembly_set` table - see similar query [here](https://github.com/EBIvariation/eva-submission/blob/61f4da73214e871a4a98cbe4f2ce5c164d09debe/eva_submission/eload_status.py#L202) for checking eload status.

Reply:
Keeping it as it is for the time being.

Checked the query and it might save one join to taxonomy (will need to join assembly anyway to get assembly_accession as assembly_set has assembly_name and not accession), but the bigger problem is that assembly_set_id is null for a lot of analyses (in thousands - especially for old ones I guess)  while vcf_reference accession is present for all except for a handful which gives me more confidence that it will give the result.

Reply:
Makes perfect sense, thanks for checking.